### PR TITLE
Add persistent session and secret

### PR DESCRIPTION
**Pull request by AndreasMeier12**
# What

Longer-lived sessions based on a persistent secret and a cookie store 

# Why

- constantly entering the password is flipping annoying, see #40

# How to see

Take care to set the `SECRET` environment variable. You can generate a suitable secret by running e.g. `dd if=/dev/urandom bs=128 count=1 | od -w128 -A n -t x1 | sed 's/ *//g'`.

- Start test server
- Log in
- Restart server
- Check to see that you're still logged in

### CODE REVIEW

File: src/bin/server.rs [hunk 1]
```diff
@@ -57,8 +57,8 @@ async fn main() {
     // `axum::response::IntoResponse`.
 
     // A closure or a function can be used as handler.
-    let store = MemoryStore::new();
-    let secret = rand::thread_rng().gen::<[u8; 128]>();
+    let store = CookieStore::new();
```
Comment (distributed) [inline src/bin/server.rs:63]: A potential future issue with `CookieStore` is that session length is limited to a total of 4093 bytes for all cookies of the domain. I think that a cookie store is an appropriate solution for this use case.

Care has to be taken that session data from an old session is not interpreted as a new session data format. This applies to all persistent cookie stores.

File: src/secret.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+use std::env;
+
+use dotenvy::dotenv;
+use hex::FromHex;
+use itertools::Itertools;
+use rand::Rng;
+
+pub fn get_secret() -> [u8; 128] {
+    dotenv().ok();
+    let raw = env::var("SECRET").ok()
```
Comment (distributed) [inline src/secret.rs:17]: I suggest adding a log line for a couple of cases:
- No `SECRET` set.
- `SECRET` has wrong length.
- `SECRET` has wrong format.

This kind of output helps greatly in troubleshooting when running the server.

**Review by distributed — CHANGES_REQUESTED** [review]

File: src/bin/server.rs (hunk 1 above)
Comment (AndreasMeier12) [reply src/bin/server.rs:63]: What would you change in this case?

File: src/secret.rs (hunk 2 above)
Comment (AndreasMeier12) [reply src/secret.rs:17]: Good idea!

Comment (distributed) [reply src/secret.rs:17]: :+1: Maybe the the log about `SECRET` not being said could just mention the command line to generate a secret from `/dev/urandom`:

```
dd if=/dev/urandom bs=128 count=1 | od -w128 -A n -t x1 | sed 's/ *//g'
```

File: src/bin/server.rs (hunk 1 above)
Comment (distributed) [reply src/bin/server.rs:63]: I think it's fine to leave it at the session store for the time being. I wanted to point out what the drawbacks are.

**Pull request merged**